yt_dlp/extractor/tbsjp.py

```diff
@@ -0,0 +1,157 @@
+            'series': 'ミライカプセル　-I have a dream-',
+            'title': 'ミライカプセル　-I have a dream-'
+        }
+    }]
+
+    def _real_extract(self, url):
+        programme_id = self._match_id(url)
+        webpage = self._download_webpage(url, programme_id)
+        meta = self._search_json(r'window\.app\s*=', webpage, 'programme info', programme_id)
+
+        programme = traverse_obj(meta, ('falcorCache', 'catalog', 'program', programme_id, 'false', 'value'))
+
+        return {
+            '_type': 'playlist',
+            'entries': [self.url_result(f'https://cu.tbs.co.jp/episode/{video_id}', TBSJPEpisodeIE, video_id)
+                        for video_id in traverse_obj(programme, ('custom_data', 'seriesList', 'episodeCode', ...))],
+            'id': programme_id,
+            **traverse_obj(programme, {
+                'categories': 'keywords',
+                'id': ('tv_episode_info', 'show_content_id', {str_or_none}),
+                'description': ('custom_data', 'program_description'),
+                'series': ('custom_data', 'program_name'),
+                'title': ('custom_data', 'program_name'),
+            }),
+        }
+
+
+class TBSJPPlaylistIE(InfoExtractor):
+    _VALID_URL = r'https://cu\.tbs\.co\.jp/playlist/(?P<id>[\da-f]+)'
+    _TESTS = [{
+        'url': 'https://cu.tbs.co.jp/playlist/184f9970e7ba48e4915f1b252c55015e',
+        'playlist_mincount': 4,
+        'info_dict': {
+            'title': 'まもなく配信終了',
+            'id': '184f9970e7ba48e4915f1b252c55015e',
+        }
+    }]
+
+    def _real_extract(self, url):
+        playlist_id = self._match_id(url)
+        page = self._download_webpage(url, playlist_id)
+        meta = self._search_json(r'window\.app\s*=', page, 'playlist info', playlist_id)
+        playlist = traverse_obj(meta, ('falcorCache', 'playList', playlist_id))
+
+        entries = []
+        for entry in traverse_obj(playlist, ('catalogs', 'value', lambda _, v: v['content_id'])):
+            # it's likely possible to get most/all of the metadata from the playlist page json instead
+            content_id = entry['content_id']
+            content_type = entry.get('content_type')
+            if content_type == 'tv_show':
+                entries.append(self.url_result(
+                    f'https://cu.tbs.co.jp/program/{content_id}', TBSJPProgramIE, content_id))
+            elif content_type == 'tv_episode':
+                entries.append(self.url_result(
+                    f'https://cu.tbs.co.jp/episode/{content_id}', TBSJPEpisodeIE, content_id))
+            else:
+                self.report_warning(f'Skipping "{content_id}" with unsupported content_type "{content_type}"')
+
+        return {
+            '_type': 'playlist',
+            'id': playlist_id,
+            **traverse_obj(playlist, {
+                'id': ('id', 'value'),
+                'title': ('display_name', 'value'),
+            }),
+            'entries': entries,
+        }
```

Comment:
if we can count on the `playlist_id` extracted from the URL as being correct, then this is much simpler/cleaner:
```suggestion
        return self.playlist_result(entries, playlist_id, traverse_obj(playlist, ('display_name', 'value')))
```